app/models/orchestration_stack_retire_request.rb

```diff
@@ -0,0 +1,14 @@
+class OrchestrationStackRetireRequest < MiqRequest
+  TASK_DESCRIPTION  = 'OrchestrationStack Retire'.freeze
+  SOURCE_CLASS_NAME = 'OrchestrationStack'.freeze
+
+  validates :request_state, :inclusion => { :in => %w(pending finished) + ACTIVE_STATES, :message => "should be pending, #{ACTIVE_STATES.join(", ")} or finished" }
+
+  validate :must_have_user
```

Comment:
@d-m-u can we add some specs for the required attributes for a RetireRequest or RetireRequestTask